### PR TITLE
Add Watrous bound for diamond norm distance

### PR DESCRIPTION
**Pull request by marcusps**
Address #60

### CODE REVIEW

**Comment by marcusps** [comment]
@karalekas Any idea what is causing the latest failure with dpkg?

**Comment by karalekas** [comment]
@marcusps Should work now, you just happened to catch me in the middle of editing the Semaphore build plan!

**Comment by marcusps** [comment]
@karalekas @kylegulshen @joshcombes Pretty minor addition, but a convenient utility to replace diamond norm distance calculations.

File: forest/benchmarking/tests/test_distance_measures.py [hunk 1]
```diff
@@ -231,6 +231,76 @@ def _gate_to_choi(gate):
...
+    # https://github.com/qutip/qutip/blob/master/qutip/tests/test_metrics.py
+    # which were in turn generated using QuantumUtils for MATLAB
+    # (https://goo.gl/oWXhO9) by Christopher Granade
+
+    _I = np.asarray([[1, 0], [0, 1]])
+    _X = np.asarray([[0, 1], [1, 0]])
+    _Y = np.asarray([[0, -1.0j], [1.0j, 0]])
+    _H = np.asarray([[1, 1], [1, -1]]) / np.sqrt(2)
+
+    def _gate_to_superop(gate):
```
Comment (karalekas) [inline forest/benchmarking/tests/test_distance_measures.py]: It looks like there is a lot of shared boilerplate between this and the `test_diamond_norm` UT, and additionally some of the functions seem very similar to some of the utilities in `superop_conversion.py`. Would it be worth consolidating some of the boilerplate and/or using some of our own utilities?

Comment (joshcombes) [reply forest/benchmarking/tests/test_distance_measures.py]: I was going to make an issue for that. Gavin wrote `diamond_norm` before we wrote the `superop_conversion` module.

Comment (marcusps) [reply forest/benchmarking/tests/test_distance_measures.py]: Yeah, I noticed that as well. TBH, we also need to revisit how to implemented the diamond norm -- there are better SDPs for it.

File: forest/benchmarking/distance_measures.py [hunk 2]
```diff
@@ -250,3 +250,20 @@ def diamond_norm(choi0: np.ndarray, choi1: np.ndarray) -> float:
...
+def watrous_bounds(choi: np.ndarray) -> float:
+    """Return the Watrous bounds for the diamon norm of a superoperator in
+    the Choi representation. If this is applied to the difference of two Choi 
+    representations, it yields bounds to the diamond norm distance.
+
+    The bound can be found in [this](https://cstheory.stackexchange.com/a/4920)
+    StackOverflow answer, although the results can also be found scattered in 
+    the literature.
+
+    :param choi: d x d matrix (for qubits, d = 4^N, where N is the number of qubits)
```
[the remark this answers is not in the document]
Comment (marcusps) [reply forest/benchmarking/distance_measures.py]: Sure we can check that d is a perfect square, and the the matrix is a square matrix.

File: forest/benchmarking/tests/test_distance_measures.py (hunk 1 above)
Comment (marcusps) [reply forest/benchmarking/tests/test_distance_measures.py]: Let's fix this in a separate PR

Comment (karalekas) [reply forest/benchmarking/tests/test_distance_measures.py]: @marcusps or @joshcombes could we open an issue?

Comment (joshcombes) [reply forest/benchmarking/tests/test_distance_measures.py]: illl open an issue now

**Pull request merged**